src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,82 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.VictorSPX;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.config.Config;
+
+import java.util.HashMap;
+import java.util.function.Supplier;
+
+public class IntakeSubsystem extends SubsystemBase {
+    /**
+     * The Singleton instance of this IntakeSubsystem. External classes should
+     * use the {@link #getInstance()} method to get the instance.
+     */
+    private final static IntakeSubsystem INSTANCE = new IntakeSubsystem();
+    
+    // The supplier of the intake speed
+    private Supplier<Double> intakeSpeed;
+    
+    // A dictionary to hold all the conditions for the intake to run
+    private HashMap<String, Boolean> conditions;
+    
+    // The intake motor (if any)
+    private VictorSPX intakeMotor;
+    
+    /**
+     * Creates a new instance of this IntakeSubsystem.
+     * This constructor is private since this class is a Singleton. External classes 
+     * should use the {@link #getInstance()} method to get the instance.
+     */
+    private IntakeSubsystem() {
+        // Initialize the private variables
+        conditions = new HashMap<>();
+        intakeSpeed = Config.INTAKE_SPEED;
+        if (Config.INTAKE_MOTOR != -1) {
+            intakeMotor = new VictorSPX(Config.INTAKE_MOTOR);
+        }
+    }
+    
+    /**
+     * Returns the Singleton instance of this IntakeSubsystem. This static method 
+     * should be used -- {@code IntakeSubsystem.getInstance();} -- by external
+     * classes, rather than the constructor to get the instance of this class.

```

Comment:
Should there be a constructor defined as private to go along with this, or is that automatic/implicit?